### PR TITLE
Support Prepatch

### PR DESCRIPTION
**Pull request by prsauer**

### CODE REVIEW

File: cloud/cors.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[
+  {
+    "origin": [
+      "http://localhost:3000",
+      "https://desktop-client.wowarenalogs.com",
```
Comment (kunchenguid) [inline cloud/cors.json:5]: might as well add in the new one here now? https://desktop.wowarenalogs.com

File: cloud/gsutil/clone.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+DONT_RUN_THIS
```
Comment (kunchenguid) [inline cloud/gsutil/clone.sh:1]: lol how about exit the script unless a -force param is set, or something like that

File: cloud/package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@wowarenalogs/cloud-functions",
```
Comment (kunchenguid) [inline cloud/package.json:2]: i constantly misremember the differences between package name and the folder name :D better to just make it @woarenalogs/cloud to be consistent with the folder structure?

File: cloud/test/test/testLogFile.txt [hunk 4]
```diff
@@ -0,0 +1,709 @@
+1/7 09:15:19.344  ZONE_CHANGE,1825,"Hook Point",0
```
Comment (kunchenguid) [inline cloud/test/test/testLogFile.txt:1]: the path "test/test/testLogFile" - might be better to make it "test/files/testLogFile"?

File: cloud/yarn.lock [hunk 5]
```diff
@@ -0,0 +1,1633 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (kunchenguid) [inline cloud/yarn.lock:1]: oh yarn. when we port into the rewrite branch we gotta remember the change to npm

File: packages/wow-combat-log-parser/package.json [hunk 6]
```diff
@@ -2,7 +2,7 @@
   "name": "wow-combat-log-parser",
   "author": "WoW Arena Logs",
   "license": "MIT",
-  "version": "4.6.4",
+  "version": "5.0.0",
```
Comment (kunchenguid) [inline packages/wow-combat-log-parser/package.json:5]: wondering if we should align the major version of this package with major wow version or combat log version :)

File: packages/wow-combat-log-parser/src/index.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import EventEmitter from 'eventemitter3';
 
-import { createShadowlandsParserPipeline } from './pipeline/shadowlands';
+import { createDragonflightParserPipeline } from './pipeline/shadowlands';
```
Comment (kunchenguid) [inline packages/wow-combat-log-parser/src/index.ts:3]: rename /pipeline/shadowlands?

File: packages/wow-combat-log-parser/src/index.ts [hunk 8]
```diff
@@ -67,13 +66,13 @@ export class WoWCombatLogParser extends EventEmitter {
       }
 
       const wowBuild = wowVersionLineMatches[2];
-      const wowVersion: WowVersion = wowBuild.startsWith('2.') ? 'tbc' : 'shadowlands';
+      const wowVersion: WowVersion = wowBuild.startsWith('2.') ? 'tbc' : 'dragonflight';
```
Comment (kunchenguid) [inline packages/wow-combat-log-parser/src/index.ts:69]: should we check wowBuild and make sure it's actually dragonflight? I bet there would still be a bunch of shadowlands log coming in over the next couple of days.

File: packages/wow-combat-log-parser/src/types.ts [hunk 9]
```diff
@@ -3,7 +3,7 @@ import { ArenaMatchStart } from './actions/ArenaMatchStart';
 import { CombatAction } from './actions/CombatAction';
 import { CombatantInfoAction } from './actions/CombatantInfoAction';
 
-export type WowVersion = 'tbc' | 'shadowlands';
+export type WowVersion = 'tbc' | 'dragonflight';
```
Comment (kunchenguid) [inline packages/wow-combat-log-parser/src/types.ts:6]: if the idea is to drop previous expansions, then maybe we should call this enum value "retail"?

**Review by kunchenguid — APPROVED** [review]

File: packages/wow-combat-log-parser/src/index.ts (hunk 8 above)
Comment (prsauer) [reply packages/wow-combat-log-parser/src/index.ts:69]: not sure what you mean? as long as the build isn't 2.*.* it'll become 'dragonflight' here

File: packages/wow-combat-log-parser/src/types.ts (hunk 9 above)
Comment (prsauer) [reply packages/wow-combat-log-parser/src/types.ts:6]: yeah i think that's probably fine, you can always ref the build # itself inside the log if you really need a version

**Pull request merged**